app/src/main/java/org/p2p/wallet/striga/offramp/models/StrigaOffRampTokenState.kt

```diff
@@ -0,0 +1,25 @@
+package org.p2p.wallet.striga.offramp.models
+
+import java.math.BigDecimal
+
+sealed class StrigaOffRampTokenState(
+    val amount: BigDecimal,
+    val balance: BigDecimal = BigDecimal.ZERO
+) {
+    class Loading(balance: BigDecimal = BigDecimal.ZERO) : StrigaOffRampTokenState(
+        amount = BigDecimal.ZERO,
+        balance = balance
+    )
+
+    class LoadingBalance(amountA: BigDecimal) : StrigaOffRampTokenState(amountA, BigDecimal.ZERO)
+
+    class Content(
```

Comment:
let's make it data class just in case

Reply:
it won't work because arguments are not a variables, these classes are just proxying data to the base class StrigaOffRampTokenState with some variation